language/functional_tests/tests/testsuite/borrow_tests/mutate_with_borrowed_loc_invalid.mvir

```diff
@@ -0,0 +1,14 @@
+module M {
+    t1() {
+        let x: u64;
+        let y: &u64;
+        x = 0;
+        y = &x;
+        x = 0;
```

Comment:
[Just thinking aloud, not suggesting any changes]

Before we introduced `Vector`, I think it would actually be safe (though probably ill-advised because it could lead to funky behavior) to allow reassignments like this. However, once you have resizable types like `Vector`, the reassignment can cause the reference to dangle and is definitely not safe (e.g., `x = Vector::singleton(1); y = Vector::borrow(0); x = Vector::empty();`).

Reply:
The code fragment in your message is not affected by the changes made in this diff.  As long as there is an outstanding reference to a local variable, it cannot be overwritten.  This diff does not change that behavior.